File: apps/android-native/app/src/main/java/com/ferni/voice/models/Persona.kt
package com.ferni.voice.models

import androidx.compose.ui.graphics.Color

/**
 * Represents a Ferni team member persona with their unique styling.
 * Colors sourced from design-system/tokens/colors.json
 */
data class Persona(
    val id: String,
    val name: String,
    val emoji: String,
    val initials: String,
    val role: String,
    val specialty: String,
    val primaryColor: Color,
    val secondaryColor: Color,
    val glowColor: Color
) {
    /** The primary hex value for gradient generation */
    val primaryHex: String
        get() = when (id) {
            "ferni" -> "#4a6741"
            "maya" -> "#a67a6a"
            "alex" -> "#5a6b8a"
            "jordan" -> "#c4856a"
            "peter" -> "#3a6b73"
            "nayan" -> "#9a7b5a"
            else -> "#4a6741"
        }

    /** Short tagline for display (alias for role) */
    val tagline: String get() = role

    companion object {
        /** Ferni - CEO & Life Coach (Sage Green) */
        val ferni = Persona(
            id = "ferni",
            name = "Ferni",
            emoji = "\uD83C\uDF3F", // 🌿
            initials = "FE",
            role = "Life Coach",
            specialty = "Leadership, life direction, bringing in the right expert",
            primaryColor = Color(0xFF4a6741),
            secondaryColor = Color(0xFF3d5a35),
            glowColor = Color(0xFF4a6741).copy(alpha = 0.4f)
        )

        /** Maya Santos - Habits Coach (Rose/Terracotta) */
        val maya = Persona(
            id = "maya",
            name = "Maya",
            emoji = "\uD83E\uDD8B", // 🦋
            initials = "MS",
            role = "Habits Coach",
            specialty = "Building habits, breaking bad ones, behavior change",
            primaryColor = Color(0xFFa67a6a),
            secondaryColor = Color(0xFF8a635a),
            glowColor = Color(0xFFa67a6a).copy(alpha = 0.4f)
        )

        /** Alex Chen - Communications Coach (Slate Blue) */
        val alex = Persona(
            id = "alex",
            name = "Alex",
            emoji = "\uD83D\uDCAC", // 💬
            initials = "AC",
            role = "Communications",
            specialty = "Difficult conversations, relationships, conflict resolution",
            primaryColor = Color(0xFF5a6b8a),
            secondaryColor = Color(0xFF4a5a73),
            glowColor = Color(0xFF5a6b8a).copy(alpha = 0.4f)
        )

        /** Jordan Taylor - Life Planner (Coral) */
        val jordan = Persona(
            id = "jordan",
            name = "Jordan",
            emoji = "\uD83D\uDCCB", // 📋
            initials = "JT",
            role = "Life Planner",
            specialty = "Goals, planning, productivity, time management",
            primaryColor = Color(0xFFc4856a),
            secondaryColor = Color(0xFFa86d55),
            glowColor = Color(0xFFc4856a).copy(alpha = 0.4f)
        )

        /** Peter John - Research Analyst (Ocean Teal) */
        val peter = Persona(
            id = "peter",
            name = "Peter",
            emoji = "\uD83D\uDD2C", // 🔬
            initials = "PJ",
            role = "Research",
            specialty = "Deep research, analysis, finding answers",
            primaryColor = Color(0xFF3a6b73),
            secondaryColor = Color(0xFF2d5359),
            glowColor = Color(0xFF3a6b73).copy(alpha = 0.4f)
        )

        /** Nayan Patel - Wisdom Sage (Warm Brown/Gold) */
        val nayan = Persona(
            id = "nayan",
            name = "Nayan",
            emoji = "\uD83E\uDDD8", // 🧘
            initials = "NP",
            role = "Wisdom",
            specialty = "Philosophy, mindfulness, deeper meaning",
            primaryColor = Color(0xFF9a7b5a),
            secondaryColor = Color(0xFF7a5b3a),
            glowColor = Color(0xFFb8956a).copy(alpha = 0.4f)
        )

        /** All personas in display order */
        val all: List<Persona> = listOf(ferni, maya, alex, jordan, peter, nayan)

        /** Get persona by ID, defaults to Ferni */
        fun get(id: String): Persona = all.find { it.id == id } ?: ferni
    }
}

File: apps/shared/Sources/FerniShared/Models/Persona.swift
import SwiftUI

// MARK: - Persona Model

/// Represents a Ferni team member persona with their unique styling
public struct Persona: Identifiable, Equatable {
    public let id: String
    public let name: String
    public let emoji: String
    public let initials: String
    public let role: String
    public let specialty: String

    // Colors from design-system/tokens/colors.json
    public let primaryColor: Color
    public let secondaryColor: Color
    public let glowColor: Color

    public init(
        id: String,
        name: String,
        emoji: String,
        initials: String,
        role: String,
        specialty: String,
        primaryColor: Color,
        secondaryColor: Color,
        glowColor: Color
    ) {
        self.id = id
        self.name = name
        self.emoji = emoji
        self.initials = initials
        self.role = role
        self.specialty = specialty
        self.primaryColor = primaryColor
        self.secondaryColor = secondaryColor
        self.glowColor = glowColor
    }

    /// The primary hex value for gradient generation
    public var primaryHex: String {
        switch id {
        case "ferni": return "#4a6741"
        case "maya": return "#a67a6a"
        case "alex": return "#5a6b8a"
        case "jordan": return "#c4856a"
        case "peter": return "#3a6b73"
        case "nayan": return "#9a7b5a"
        default: return "#4a6741"
        }
    }

    /// Short tagline for display (alias for role)
    public var tagline: String {
        role
    }
}

// MARK: - Persona Registry

/// All available Ferni team personas
/// Colors sourced from design-system/tokens/colors.json
public enum PersonaRegistry {

    /// Ferni - CEO & Life Coach (Sage Green)
    public static let ferni = Persona(
        id: "ferni",
        name: "Ferni",
        emoji: "🌿",
        initials: "FE",
        role: "Life Coach",
        specialty: "Leadership, life direction, bringing in the right expert",
        primaryColor: Color(hex: 0x4a6741),
        secondaryColor: Color(hex: 0x3d5a35),
        glowColor: Color(hex: 0x4a6741).opacity(0.4)
    )

    /// Maya Santos - Habits Coach (Rose/Terracotta)
    public static let maya = Persona(
        id: "maya",
        name: "Maya",
        emoji: "🦋",
        initials: "MS",
        role: "Habits Coach",
        specialty: "Building habits, breaking bad ones, behavior change",
        primaryColor: Color(hex: 0xa67a6a),
        secondaryColor: Color(hex: 0x8a635a),
        glowColor: Color(hex: 0xa67a6a).opacity(0.4)
    )

    /// Alex Chen - Communications Coach (Slate Blue)
    public static let alex = Persona(
        id: "alex",
        name: "Alex",
        emoji: "💬",
        initials: "AC",
        role: "Communications",
        specialty: "Difficult conversations, relationships, conflict resolution",
        primaryColor: Color(hex: 0x5a6b8a),
        secondaryColor: Color(hex: 0x4a5a73),
        glowColor: Color(hex: 0x5a6b8a).opacity(0.4)
    )

    /// Jordan Taylor - Life Planner (Coral)
    public static let jordan = Persona(
        id: "jordan",
        name: "Jordan",
        emoji: "📋",
        initials: "JT",
        role: "Life Planner",
        specialty: "Goals, planning, productivity, time management",
        primaryColor: Color(hex: 0xc4856a),
        secondaryColor: Color(hex: 0xa86d55),
        glowColor: Color(hex: 0xc4856a).opacity(0.4)
    )

    /// Peter John - Research Analyst (Ocean Teal)
    public static let peter = Persona(
        id: "peter",
        name: "Peter",
        emoji: "🔬",
        initials: "PJ",
        role: "Research",
        specialty: "Deep research, analysis, finding answers",
        primaryColor: Color(hex: 0x3a6b73),
        secondaryColor: Color(hex: 0x2d5359),
        glowColor: Color(hex: 0x3a6b73).opacity(0.4)
    )

    /// Nayan Patel - Wisdom Sage (Warm Brown/Gold)
    public static let nayan = Persona(
        id: "nayan",
        name: "Nayan",
        emoji: "🧘",
        initials: "NP",
        role: "Wisdom",
        specialty: "Philosophy, mindfulness, deeper meaning",
        primaryColor: Color(hex: 0x9a7b5a),
        secondaryColor: Color(hex: 0x7a5b3a),
        glowColor: Color(hex: 0xb8956a).opacity(0.4)
    )

    /// All personas in display order
    public static let all: [Persona] = [ferni, maya, alex, jordan, peter, nayan]

    /// Get persona by ID
    public static func get(_ id: String) -> Persona {
        all.first { $0.id == id } ?? ferni
    }
}
